packages/react-cosmos/src/linked-list.js

```diff
@@ -0,0 +1,12 @@
+export default function createLinkedList(items) {
+  function advanceList(toIndex) {
+    return {
+      value: items[toIndex],
+      next: function getNextItem(nextIndex) {
+        return advanceList(nextIndex);
+      }.bind(null, toIndex + 1),
```

Comment:
Tried using generators, but realised they work different. Tried my best to explain in this commit message: 3a02cfbe06d55ce39a6ec97bf1c9b3bf56dacd7e